tests/tests_pytorch/test_cli.py

```diff
@@ -1860,6 +1860,34 @@ def test_lightning_cli_args_and_sys_argv_warning():
         LightningCLI(TestModel, run=False, args=["--model.foo=789"])
 
 
+def test_add_class_args_required_false_skips_addition(tmp_path):
+    from lightning.pytorch import callbacks, cli
+
+    class FooCheckpoint(callbacks.ModelCheckpoint):
+        def __init__(self, dirpath, *args, **kwargs):
+            super().__init__(dirpath, *args, **kwargs)
+
+    class SimpleModel:
+        def __init__(self):
+            pass
+
+    class SimpleDataModule:
+        def __init__(self):
+            pass
```

Comment:
```suggestion
```
`LightningCLI` requires modules to inherit from the lightning classes. These don't so there is no way this would work. Also there is no need to implement new classes, that is what `BoringModel` and `BoringDataModule` are for.